reconcile/utils/oc.py

```diff
@@ -620,6 +628,177 @@ def _run_json(self, cmd, allow_not_found=False):
         return out_json
 
 
+class OC(OCDeprecated):
+    def __init__(self, server, token, jh=None, settings=None,
+                 init_projects=False, init_api_resources=False,
+                 local=False):
+        super().__init__(server, token, jh, settings,
+                         init_projects, False,
+                         local)
+        self.client = self._get_client(server, token)
+        self.api_kind_version = self.get_api_resources()
+        self.object_clients = {}
+        self.init_api_resources = init_api_resources
+        if self.init_api_resources:
+            self.api_resources = self.api_kind_version.keys()
+        else:
+            self.api_resources = None
+
+    def _get_client(self, server, token):
+        opts = dict(
+            api_key={'authorization': f'Bearer {token}'},
+            host=server,
+            verify_ssl=True
+        )
+
+        if self.jump_host is not None:
+            opts['proxy'] = f'https://{self.jump_host.get_hostname()}:8888'
```

Comment:
This is not something already in place, right? I mean, I can't test this now?

Reply:
It is and it isn't. :) I had a proxy setup where access was controlled via allow clauses, and it was configured to work from my laptop.  That won't scale or manage well, so I am working on changing the proxy to use client certs for authentication.

Reply:
@apahim This should now be fully testable now.